compute_tools/src/compute.rs

```diff
@@ -343,4 +345,71 @@ impl ComputeNode {
         self.prepare_pgdata()?;
         self.run()
     }
+
+    // Look for core dumps and collect backtraces.
+    //
+    // EKS worker nodes have following core dump settings:
+    //   /proc/sys/kernel/core_pattern -> core
+    //   /proc/sys/kernel/core_uses_pid -> 1
+    //   ulimint -c -> unlimited
+    // which results in core dumps being written to postgres data directory as core.<pid>.
+    //
+    // For local testing I also added support for lldb, which writes core dumps to /cores.
+    fn check_for_core_dumps(&self) {
+        let core_dump_dir = if std::env::consts::OS == "macos" {
+            std::path::Path::new("/cores/")
+        } else {
+            std::path::Path::new(&self.pgdata)
+        };
+
+        info!(
+            "checking for core dumps in {}",
+            core_dump_dir.to_string_lossy()
+        );
+
+        let paths = fs::read_dir(core_dump_dir).expect("cannot list pgdata for core dumps");
+        for path in paths {
+            if let Some(name) = path
+                .expect("can't read pgdata directory entry")
+                .path()
+                .file_name()
+            {
+                let name = name.to_string_lossy();
+                if name.starts_with("core.") {
+                    warn!("core dump found: {}, collecting backtrace", name);
+                    let full_core_path = std::path::Path::new(core_dump_dir).join(name.as_ref());
+                    let full_core_path = full_core_path.to_string_lossy();
+
+                    let backtrace = if std::env::consts::OS == "macos" {
+                        Command::new("lldb")
+                            .args(&[
+                                "-c",
+                                &full_core_path,
+                                "--batch",
+                                "-o",
+                                "thread backtrace all",
+                                "-o",
+                                "quit",
+                            ])
+                            .output()
+                            .expect("cannot start lldb process")
+                    } else {
+                        Command::new("gdb")
+                            .args(&["--batch", "-q", &self.pgbin, &full_core_path, "-ex", "bt"])
```

Comment:
I think file should be passed via `--core`.

Reply:
```
# echo 'int main(){*(volatile int *)0=42;}' > x.c
# gcc -g x.c
# ./a.out 
Segmentation fault (core dumped)
# gdb --batch -q ./a.out ./core.3996 -ex bt
[New LWP 3996]
Core was generated by `./a.out'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000000400480 in main () at x.c:1
1	int main(){*(volatile int *)0=42;}
#0  0x0000000000400480 in main () at x.c:1
```

man also mentions syntax `gdb [OPTIONS] [prog|prog procID|prog core] `